rskj-core/src/main/java/co/rsk/mine/BlockToMineBuilder.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.mine;
+
+import co.rsk.config.MiningConfig;
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.Coin;
+import co.rsk.core.DifficultyCalculator;
+import co.rsk.core.RskAddress;
+import co.rsk.core.bc.BlockExecutor;
+import co.rsk.core.bc.FamilyUtils;
+import co.rsk.remasc.RemascTransaction;
+import co.rsk.validators.BlockValidationRule;
+import org.apache.commons.collections4.CollectionUtils;
+import org.ethereum.core.*;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.ReceiptStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.math.BigInteger;
+import java.time.Clock;
+import java.util.*;
+
+/**
+ * This component helps build a new block to mine.
+ * It can also be used to generate a new block from the pending state, which is useful
+ * in places like Web3 with the 'pending' parameter.
+ */
+@Component
+public class BlockToMineBuilder {
+    private static final Logger logger = LoggerFactory.getLogger("blocktominebuilder");
+
+    private final MiningConfig miningConfig;
+    private final Repository repository;
+    private final BlockStore blockStore;
+    private final PendingState pendingState;
+    private final DifficultyCalculator difficultyCalculator;
+    private final GasLimitCalculator gasLimitCalculator;
+    private final BlockValidationRule validationRules;
+
+    private final Clock clock;
+    private final MinimumGasPriceCalculator minimumGasPriceCalculator;
+    private final MinerUtils minerUtils;
+    private final BlockExecutor executor;
+
+    private final RskAddress coinbaseAddress;
```

Comment:
coinbaseAddress is referenced only once in this class, private field does not seem to be necessary now.